javascript-bank/account.js

```diff
@@ -0,0 +1,43 @@
+/* exported Account */
+function Account(number, holder) {
+  this.number = number;
+  this.holder = holder;
+  this.transactions = [];
+}
+
+Account.prototype.deposit = function (amount) {
+  if ((amount <= 0) || (typeof amount !== 'number') || (isNaN(amount)) || (!(Number.isInteger(amount)))) {
+    return false;
+  } else if (amount > 0) {
+    var newTransaction = new Transaction('deposit', amount);
+    this.transactions.push(newTransaction);
+    return true;
+  }
+};
+
+Account.prototype.withdraw = function (amount) {
+  if ((amount <= 0) || (typeof amount !== 'number') || (isNaN(amount)) || (!(Number.isInteger(amount)))) {
+    return false;
+  } else if (amount > 0) {
+    var newTransaction = new Transaction('withdrawal', amount);
+    this.transactions.push(newTransaction);
+    return true;
+  }
+};
+
+Account.prototype.getBalance = function () {
+  if (this.transactions.length === 0) {
+    return 0;
+  } else {
```

Comment:
This conditional is not necessary. In your `else` block, you start the variable at `0` and then `return` that value after the loop. If there are no items stored in the `this.transactions`, the `length` would be `0`, so the loop would never start, and you would still return `0` for an empty transaction list.